numpy/core/tests/test_umath.py

```diff
@@ -1767,11 +1771,21 @@ def __array_ufunc__(self, func, method, *inputs, **kwargs):
         # With 1 NotImplemented
         res = np.multiply(c, a)
         assert_equal(res, "A")
+        assert_equal(c.count, 1)
 
         # Both NotImplemented.
+        c = C()
+        c_sub = CSub()
         assert_raises(TypeError, np.multiply, c, c_sub)
+        assert_equal(c.count, 1)
+        assert_equal(c_sub.count, 1)
         assert_raises(TypeError, np.multiply, c_sub, c)
+        assert_equal(c.count, 2)
+        assert_equal(c_sub.count, 2)
+        assert_raises(TypeError, np.multiply, c, c)
+        assert_equal(c.count, 3)
         assert_raises(TypeError, np.multiply, 2, c)
+        assert_equal(c.count, 4)
```

Comment:
Test might be clearer and more reorderable if you did `c.count = c_sub.count = 0` each time

Reply:
OK, does make more sense. Done.